calblueprint/siren-web
# Permissions WIP

# ✨ New in this PR ✨
Siren User permissions!

## One liner: what did you do? 🚨 

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [X] Gate login behind permissions 
 - [X] Create permissions table
 - [x] Create approval table
 - [x] Gate features behind permissions 

## How can the reviewer test your code? 👩‍💻
_example: run `npm start` and click through Navigation bar_ 

1. run `npm run dev` and sign out of any account you're currently in.
2. Register an account
3. Try logging into said account (it should say need access)
4. Go to firebase into collections `sirenUsers`, find the newly made account, and change `isApproved` to "Approved" and role to "Super"
5. Try logging in again (should be good)
7. Click user settings/profile at top right of page
8. Go to Permissions tab and change around the values of `Role/Permission` and verify they change in Firebase as well
9. Try signing up with another account and approving it/denying it from the Super admin account you just made.

## Any bugs you encountered or still having trouble with? 🐛
_describe how you solved them such that someone can follow your explanation_ 


## Resources 📔
_link online resources and related PRs_ 


🧜‍♀️ cc: @shannonbonet @phoebeli23
